### ISSUE TITLE
'operation timed out' adding single large package

### ISSUE THREAD

**Issue by james-vincent** [prefix-dev/pixi#1095]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://github.com/prefix-dev/pixi/releases) of pixi, using `pixi --version`.

### Reproducible example

```shell

 $ ./pixi add pytorch-cuda
  × failed to download package https://conda.anaconda.org/pytorch/linux-64/pytorch-2.2.2-py3.11_cuda12.1_cudnn8.9.2_0.tar.bz2
  ├─▶ an io error occurred
  ├─▶ failed to unpack `/nfs/builders/sbgrid/.cache/rattler/cache/pkgs/pytorch-2.2.2-py3.11_cuda12.1_cudnn8.9.2_0/lib/
  │   python3.11/site-packages/torch/include/torch/csrc/jit/tensorexpr/cpp_codegen.h`
  ├─▶ failed to unpack `lib/python3.11/site-packages/torch/include/torch/csrc/jit/tensorexpr/cpp_codegen.h` into `/nfs/builders/
  │   sbgrid/.cache/rattler/cache/pkgs/pytorch-2.2.2-py3.11_cuda12.1_cudnn8.9.2_0/lib/python3.11/site-packages/torch/include/
  │   torch/csrc/jit/tensorexpr/cpp_codegen.h`
  ├─▶ request or response body error: operation timed out
  ╰─▶ operation timed out


 $ ./pixi --version
pixi 0.17.1

 $ cat /etc/redhat-release 
CentOS Linux release 7.9.2009 (Core)


```

### Issue description

pixi add command gives timeout error while uncompressing large package. Repeating command gives same error but shows a stop in the uncompress at different points (different files named).

### Expected behavior

Install should succeed. Manually downloading the same package file and uncompressing works fine, no issues.

Update: Set PIXI_CACHE_DIR to a local SSD and operation completes. Seems like there is simply a timeout value for the process that isn't long enough. Can it be set by user?

**Comment by baszalmstra** [comment]
I think by default we can increase this time-out to 10 minutes?

**Comment by baszalmstra** [comment]
@james-vincent What do you think a good default timeout value would be? Currently it was set to 5 minutes.

**Comment by wolfv** [comment]
Isn't it weird that it times out while it's (probably) still downloading?

**Comment by james-vincent** [comment]
It depends on what is being done within the timeout period. It seems like it is much more than just unpacking a single package. Is it the whole resolve/install step? In that case it should be at least ten minutes.

When timeout occurs, an error message stating that you can set the timeout value would be very helpful.

Jim

**Comment by ekvidera** [comment]
New contributor here, I'd love to take a crack at this.  It should be fairly straightforward to
- move the timeout to the global config
- include a user-friendly message on timeout
- possibly update to 10 minutes rather than 5?

**Comment by ruben-arts** [comment]
@ekvidera Yes please!

**Comment by james-vincent** [comment]
> @ekvidera Yes please!

Second this.

**Comment by wolfv** [comment]
Fixed in #1329
